readers/hdfs/src/test/java/com/criteo/hadoop/garmadon/hdfs/hive/HiveClientTest.java

```diff
@@ -0,0 +1,179 @@
+            assertTrue(e.getMessage().contains("Table not found not_a_table"));
+        }
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void tryCreateTableWithUnsupportedDataTypeShouldThrowRuntimeException() throws SQLException {
+        PrimitiveType unsupported = new PrimitiveType(Type.Repetition.OPTIONAL, PrimitiveType.PrimitiveTypeName.INT96, "unsupported");
+
+        MessageType schema = new MessageType("fs", unsupported);
+
+        String table = "fs";
+        String location = "file:" + hdfsTemp + "/garmadon_database/fs";
+        HiveClient hiveClient = new HiveClient(driverName, "jdbc:hive2://localhost:" + port, "garmadon",
+            hdfsTemp + "/garmadon_database");
+        hiveClient.createTableIfNotExist(table, schema, location);
+    }
+
+    @Test
+    public void shouldProvideHiveTypeFromParquetType() throws Exception {
+        HiveClient hiveClient = new HiveClient(driverName, "jdbc:hive2://localhost:" + port, "garmadon",
+            hdfsTemp + "/garmadon_database");
+
+        PrimitiveType string = new PrimitiveType(Type.Repetition.OPTIONAL, PrimitiveType.PrimitiveTypeName.BINARY, "name");
+        assertEquals("string", hiveClient.inferHiveType(string));
+
+        PrimitiveType array_string = new PrimitiveType(Type.Repetition.REPEATED, PrimitiveType.PrimitiveTypeName.BINARY, "name");
+        assertEquals("array<string>", hiveClient.inferHiveType(array_string));
+
+        PrimitiveType int32 = new PrimitiveType(Type.Repetition.OPTIONAL, PrimitiveType.PrimitiveTypeName.INT32, "name");
+        assertEquals("int", hiveClient.inferHiveType(int32));
+
+        PrimitiveType int64 = new PrimitiveType(Type.Repetition.OPTIONAL, PrimitiveType.PrimitiveTypeName.INT64, "name");
+        assertEquals("bigint", hiveClient.inferHiveType(int64));
+
+        PrimitiveType floatz = new PrimitiveType(Type.Repetition.OPTIONAL, PrimitiveType.PrimitiveTypeName.FLOAT, "name");
+        assertEquals("float", hiveClient.inferHiveType(floatz));
+
+        PrimitiveType doublez = new PrimitiveType(Type.Repetition.OPTIONAL, PrimitiveType.PrimitiveTypeName.DOUBLE, "name");
+        assertEquals("double", hiveClient.inferHiveType(doublez));
+
+        PrimitiveType booleanz = new PrimitiveType(Type.Repetition.OPTIONAL, PrimitiveType.PrimitiveTypeName.BOOLEAN, "name");
+        assertEquals("boolean", hiveClient.inferHiveType(booleanz));
+    }
+
+
+    @Test(expected = Exception.class)
+    public void shouldThrowExceptionForUnknownParquetType() throws Exception {
+        HiveClient hiveClient = new HiveClient(driverName, "jdbc:hive2://localhost:" + port, "garmadon",
+            hdfsTemp + "/garmadon_database");
+
+        PrimitiveType unsupported = new PrimitiveType(Type.Repetition.OPTIONAL, PrimitiveType.PrimitiveTypeName.INT96, "unsupported");
+        hiveClient.inferHiveType(unsupported);
+    }
+
+    @Test
+    public void shouldReconnectOnTTransportException() throws Exception {
```

Comment:
Typo: OnTransport

Reply:
Plus I don't get the test functionality. I would expect that it fails if the server is stopped, then successfully reconnects and is able to perform operations once reconnected.

Reply:
Not a typo the exception is TTransportException

Reply:
The test check that we try to reconnect